spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-basic-sample/pom.xml

```diff
@@ -39,7 +38,7 @@
         <dependency>
             <groupId>org.springframework.shell</groupId>
             <artifactId>spring-shell-starter</artifactId>
-            <version>2.1.4</version>
+            <version>3.0.0-M1</version>
```

Comment:
A reminder to switch once released.
Just in case `spring-shell-starter:3.0.0` release gets to be our blocker on release, is 2.1.4 compatible with spring boot 3 at all, or do you need to switch off of this dependency?

Reply:
The 2.1.4 is not compatible with Spring Boot 3 as the test failed so I have to upgrade the version.

The thing is when `spring-shell-starter:3.0.0` releases, I have to change the test accordingly because spring-shell changed the API visualibility ([`evaluate` method](https://github.com/spring-projects/spring-shell/blob/main/spring-shell-core/src/main/java/org/springframework/shell/Shell.java#L193) becomes private). I have to figure it out before our release.

Reply:
One option if this becomes blocker is to move away from `spring-shell` for this sample to either web-based like [firestore-sample](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/tree/main/spring-cloud-gcp-samples/spring-cloud-gcp-data-firestore-sample), or remove the interactive aspect like the other [datastore-sample](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/tree/main/spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-sample). Benefit is that we remove dependency on `shell` module for the project, which may be easier for future maintenance.
But that's more work at the moment, so I would only consider if this actually blocks you.

Reply:
Thanks for the suggestion.

Spring shell doesn't have a release date yet, so maybe I need to change the sample to remove the shell dependency. I'll look into it next week.

Reply:
I've changed the sample to web-based application.